lib/astutils.cpp

```diff
@@ -238,6 +238,33 @@ bool isSameExpression(bool cpp, bool macro, const Token *tok1, const Token *tok2
     return commutativeEquals;
 }
 
+bool isContainerYield(const Token * const cond, Library::Container::Yield y, std::string name)
+{
+    if(!cond)
+        return false;
+    if (cond->str() == "(") {
+        const Token* tok = cond->astOperand1();
+        if(tok && tok->str() == ".") {
+            if(tok->astOperand1() && tok->astOperand1()->valueType()) {
+                if(const Library::Container *container = tok->astOperand1()->valueType()->container) {
+                    return tok->astOperand2() && y == container->getYield(tok->astOperand2()->str());
+                }
+            }
+            else if(!name.empty()) {
+                return Token::simpleMatch(cond, "( )") && cond->previous()->str() == name;
+            }
+        }
+    }
+    return false;
+}
+
+bool equalTokValue(const Token * const tok1, const Token * const tok2)
```

Comment:
this name is confusing. as I understand it, this function returns true if both tokens have values and those are completely different?

I am a bit worried for false positives. It would feel better if you checked instead that the tokens have Known values that are different. Do you see a problem with that?

Reply:
> I am a bit worried for false positives. It would feel better if you checked instead that the tokens have Known values that are different. Do you see a problem with that?

hmm.. maybe looking at the 'Possible' values also is ok to start with. We can try it and see if there are false positives.

Reply:
> this name is confusing. as I understand it, this function returns true if both tokens have values and those are completely different?

It returns true if the token is one of the container yield methods. The method only takes one token.

> I am a bit worried for false positives. It would feel better if you checked instead that the tokens have Known values that are different.

What do you mean?

Reply:
I understand what you mean. It was pointing at a different line on github so I was confused.

> this name is confusing. as I understand it, this function returns true if both tokens have values and those are completely different?

No, it supposed to returns true if `tok1` contains one of the same values that `tok2` has.  Although, it might be better to check that each has the same set of possible value. Let me fix that.